x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/GroupConfig.java

```diff
@@ -150,6 +153,11 @@ public static GroupConfig fromXContent(final XContentParser parser, boolean leni
 
             ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, parser::getTokenLocation);
             String destinationFieldName = parser.currentName();
+            if (!validAggMatcher.reset(destinationFieldName).matches()) {
+                throw new ParsingException(parser.getTokenLocation(), "Invalid group name [" + destinationFieldName
+                        + "]. Group names can contain any character except '[', ']', and '>'");
```

Comment:
Please also mention "^".

Reply:
Please disregard this comment, I've just read @droberts195 comment that we should not ban "^".